### PR TITLE
LPD-34383 - Account group api

### PR DESCRIPTION
**Pull request by matyaswollner**
https://liferay.atlassian.net/browse/LPD-34383

cc @smottal

### CODE REVIEW

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountGroupResourceImpl.java [hunk 1]
```diff
@@ -361,6 +389,27 @@ private AccountGroup _toAccountGroup(
 			_getDTOConverterContext(accountGroup.getAccountGroupId()));
 	}
 
+	private AccountGroup _updateAccountGroup(
+			AccountGroup accountGroup,
+			com.liferay.account.model.AccountGroup serviceAccountGroup)
+		throws Exception {
+
+		_accountGroupService.updateAccountGroup(
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountGroupResourceImpl.java:397]: serviceAccountGroup = _acc...

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountGroupResourceImpl.java [hunk 2]
```diff
@@ -361,6 +389,27 @@ private AccountGroup _toAccountGroup(
...
+				serviceAccountGroup.getDescription()),
+			GetterUtil.getString(
+				accountGroup.getName(), serviceAccountGroup.getName()),
+			_createServiceContext(accountGroup));
+
+		serviceAccountGroup = _accountGroupService.updateExternalReferenceCode(
+			serviceAccountGroup.getAccountGroupId(),
+			accountGroup.getExternalReferenceCode());
+
+		return _toAccountGroup(serviceAccountGroup);
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountGroupResourceImpl.java:410]: inline these

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by smottal** [comment]
@matyaswollner can you please add also an integration test for the patch method and for the put where you don't pass the name?
The patch should works fine and the previous name should be equal to the returned one
The put should fail with the message `The account group name is not valid` (or something like that)

**Comment by matyaswollner** [comment]
ci:test:sf

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java [hunk 3]
```diff
@@ -241,6 +282,101 @@ private AccountGroup _postAccountGroup(AccountGroup accountGroup)
...
+	}
+
+	private void _testPutAccountGroupByExternalReferenceWithoutName()
+		throws Exception {
+
+		AccountGroup postAccountGroup = _postAccountGroup(randomAccountGroup());
+
+		AccountGroup randomAccountGroup = randomAccountGroup();
+
+		randomAccountGroup.setName((String)null);
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java:344]: same here

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java [hunk 4]
```diff
@@ -241,6 +282,101 @@ private AccountGroup _postAccountGroup(AccountGroup accountGroup)
...
+
+		try {
+			accountGroupResource.putAccountGroupByExternalReferenceCode(
+				postAccountGroup.getExternalReferenceCode(),
+				randomAccountGroup);
+			Assert.fail();
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java:350]: add an empty line before

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java [hunk 5]
```diff
@@ -241,6 +282,101 @@ private AccountGroup _postAccountGroup(AccountGroup accountGroup)
...
+				"The account group name is invalid", problem.getTitle());
+		}
+	}
+
+	private void _testPutAccountGroupWithoutName() throws Exception {
+		AccountGroup postAccountGroup = _postAccountGroup(randomAccountGroup());
+
+		AccountGroup randomAccountGroup = randomAccountGroup();
+
+		randomAccountGroup.setName((String)null);
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java:365]: same here

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java [hunk 6]
```diff
@@ -241,6 +282,101 @@ private AccountGroup _postAccountGroup(AccountGroup accountGroup)
...
+
+		try {
+			accountGroupResource.putAccountGroup(
+				postAccountGroup.getId(), randomAccountGroup);
+			Assert.fail();
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java:370]: add an empty line before this

**Review by smottal — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java (hunk 4 above)
Comment (matyaswollner) [reply modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountGroupResourceTest.java:350]: Before the Assert.Fail or before line 347?

**Comment by matyaswollner** [comment]
ci:test:sf

**Review by smottal — APPROVED** [review]

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Review by smottal — APPROVED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Comment by matyaswollner** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:report:27081021

**Comment by smottal** [comment]
ci:forward:force

**Pull request closed without merging**